private/bufpkg/bufcheck/buflint/buflint_test.go

```diff
@@ -986,6 +1064,48 @@ func testLintWithModifiers(
 	)
 	require.NoError(t, err)
 
+	var validateReadWriteBucket storage.ReadWriteBucket
+	if dependencyPathPrefix != "" {
+		validateReadWriteBucket, err = storageosProvider.NewReadWriteBucket(
+			"testdata",
+			storageos.ReadWriteBucketWithSymlinksIfSupported(),
+		)
+		require.NoError(t, err)
+
+		// Copy the testdata directory to a temporary directory
```

Comment:
Why are we copying it? This should just be a read-only thing, we should be able to just make a union bucket via `storage.MultiReadBucket` between the `readWriteBucket` and the additional dependency, and then pass this union bucket to `BuildForBucket`.

If we did/do need to copy, there is `storage.Copy`.

Reply:
resolved in [1b2c71c](https://github.com/bufbuild/buf/pull/2377/commits/1b2c71cceeb54702ea58bc69970a1a434fb6410e)